runtime/gc_modron_startup/mminit.cpp

```diff
@@ -2959,6 +2959,10 @@ gcInitializeDefaults(J9JavaVM* vm)
 	}
 #endif /* OMR_GC_CONCURRENT_SCAVENGER */
 
+#ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
+	extensions->getArrayHeaderSize =  ((MM_GCExtensionsBase *)vm->gcExtensions)->indexableObjectModel.getHeaderSize();
```

Comment:
Can you compile this on all platforms, Im not sure all compilers generate outline-of-line versions of inline functions.